Js playercardcounters

jacobstringer opened the pull request.

jacobstringer committed:
Changes Cardhist serialise and the print_json function to send back card history information of every player.

jacobstringer committed:
Includes card colour counters in player object for front end display, and ignores any cards used for wonders.

jacobstringer commented:
Changes Cardhist serialise and the print_json function to send back card history information of every player.

I will let @Marthijn-B merge this when he's happy that the front end is updated so this doesn't break anything.

Also now includes card colour counters.

@mmdonaldson, could you create a migration file for this PR and commit it? I'm still not able to do it.

mmdonaldson reviewed (changes requested):
I will add the db migration script tonight. Blocking this PR until then incase we accidentally merge

jacobstringer committed:
Includes card_colour in the card history objects for some front end post-processing.

Marthijn-B committed:
CardHistory Extraction

PlayerDisplay.js finds correct card History list for currently displayed wonder.
Passes that data to CardHist.js to render

Marthijn-B commented:
Did my PR fail because there were alot of changes to the file?
ie, this branch is behind the master branch atm. 
This works on my machine... hmmm

jacobstringer commented:
@Marthijn-B, it looks like the test failed because there's now no ./CardHist in PlayerDisplay.js. It'll require a bit of an update on that test I would say.

Marthijn-B commented:
Ok, I see how to find that info now. Thanks.
You reckon I could just upload my CardHist.js file here, would fix that? (even though it has changed a little from the master branch now?

mmdonaldson committed:
Add the relevant db migration script

mmdonaldson commented:
@Marthijn-B if I was you i'd merge master into your branch or rebase

Marthijn-B commented:
What I used, was the current master (as of this afternoon)., added Jacobs changed files from this PR, and worked off that. 
Currently this PR status: "This branch is 5 commits ahead, 11 commits behind master."
CardHist.js in already in the master. (entered GitHub after this PR was created)
#Slightly confused

mmdonaldson commented:
@Marthijn-B yeap it gets a bit chaotic when multiple people work on a branch. Once person will have to clean it all up

mmdonaldson commented:
@Marthijn-B @jacobstringer happy to do it but just let me because only person can work on the branch to clean it up for merge

jacobstringer commented:
Feel free to sort as needs be. I won't be doing any more back end work on this branch... it's mainly a case of getting the front end to be happy with the changes.



Sent from my Samsung Galaxy smartphone.

-------- Original message --------
From: Mitchell Donaldson <notifications@github.com>
Date: 3/10/17 11:13 PM (GMT+12:00)
To: Massey-7-Wonders-project-team/Massey_Project_7_Wonders <Massey_Project_7_Wonders@noreply.github.com>
Cc: jacobstringer <jacobstringer@windowslive.com>, Mention <mention@noreply.github.com>
Subject: Re: [Massey-7-Wonders-project-team/Massey_Project_7_Wonders] Js playercardcounters (#129)


@Marthijn-B<https://github.com/marthijn-b> @jacobstringer<https://github.com/jacobstringer> happy to do it but just let me because only person can work on the branch to clean it up for merge

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Massey-7-Wonders-project-team/Massey_Project_7_Wonders/pull/129#issuecomment-333799236>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AcXeHN_0ECaSetlB6OtR3TB1_suMSTIJks5sogjigaJpZM4PrcL4>.

Marthijn-B commented:
All good. I wont do anymore to this branch

Marthijn-B committed:
Cascade Card History Display

Grouped together by color
Individual rows

Marthijn-B committed:
Merge branch 'master' into JS-playercardcounters

Marthijn-B commented:
So I added CardHist.js into this PR.
Created conflicts in both CardHist and PlayerDisplay
Resolved all conflicts for these two files to the upto date code.
Lets see if circleci passes...

Marthijn-B committed:
Removed =====

Missed one of these lines when resolving conflicts

Marthijn-B committed:
Needed missed some double up code

mmdonaldson commented on static/src/components/CardHist.js:
```diff
@@ -1,48 +1,88 @@
+                                let marginLeft = -60;
+                                if (index === 0) { marginLeft = 0; }
+                                const topMargin = index * 32;
+                                console.log(index, marginLeft);
```
Can we remove console.log

Marthijn-B replied:
Always one still slips thru. Removed the rest.

mmdonaldson commented on static/src/components/CardHist.js:
```diff
@@ -1,48 +1,88 @@
+    let green = [];
+    let red = [];
+
+    for (let i = 1; i < unsorted.length; ++i) {
```
Does the index of history start at 1?

Marthijn-B replied:
Starts at 1 because index zero is the wonder board itself. No need to display that, just the cards

Marthijn-B committed:
Removed remaining console log

Marthijn-B reviewed (commented):
CardHist.js changed

Marthijn-B committed:
Check whole list

Even though we can assume that the Wonder Board is in position zero, we should still check that position incase the order changes for some reason. If the card_color === 'wonder', the loop will do nothing with that card anyway.

Marthijn-B reviewed (commented):
Loop to start at index zero
